Source/WebCore/platform/graphics/Path.cpp

```diff
@@ -628,8 +628,20 @@ bool Path::hasSubpaths() const
 
 FloatRect Path::fastBoundingRect() const
 {
-    if (auto* segment = asSingle())
-        return PathStream::computeFastBoundingRect(singleElementSpan(*segment));
+    if (auto* segment = asSingle()) {
+        FloatPoint currentPoint;
+        FloatPoint lastMoveToPoint;
+
+        auto boundingRect = FloatRect::smallestRect();
+        segment->extendFastBoundingRect(currentPoint, lastMoveToPoint, boundingRect);
+
+        if (boundingRect.isSmallest()) {
+            currentPoint = segment->calculateEndPoint(currentPoint, lastMoveToPoint);
+            boundingRect.extend(currentPoint);
+        }
+
+        return boundingRect;
```

Comment:
Can we put this in a `PathSegment::fastBoundingRect` (or other name like `fastBoundingRectForSingleSegment`), just to keep these `Path` methods that switch based on their backing data type simple?